modules/database/src/std/rec/histogramRecord.c

```diff
@@ -140,9 +140,9 @@ static long wdogInit(histogramRecord *prec)
     if (!prec->wdog && prec->sdel > 0) {
         /* initialize a callback object */
         pcallback = calloc(1, sizeof(myCallback));
-        pcallback->prec = prec;
         if (!pcallback)
             return -1;
+        pcallback->prec = prec;
```

Comment:
Definitely a bug, going back over 15 years. I am manually applying a version of this fix to the 3.15 branch (we fix bugs in the earliest supported branch that contains them).

Reply:
cf. 41f1b0ffb51b611f4290dfcc9454b8b72dedbc81